qiskit_nature/transformers/active_space_transformer.py

```diff
@@ -56,31 +56,66 @@ class ActiveSpaceTransformer(BaseTransformer):
+                 active_orbitals: Optional[List[int]] = None,
+                 freeze_core: bool = False,
+                 remove_orbitals: Optional[List[int]] = None,
+                 ):
         """Initializes a transformer which can reduce a `QMolecule` to a configured active space.
 
+        This transformer requires the AO-basis matrices `hcore` and `eri` to be available, as well
+        as the basis-transformation matrix `mo_coeff`. A `QMolecule` produced by Qiskit's drivers in
+        general satisfies these conditions unless it was read from an FCIDump file. However, those
+        integrals are likely already reduced by the code which produced the file or can be
+        transformed using this driver after copying the MO-basis integrals of the produced
+        `QMolecule` into the AO-basis containers and initializing `mo_coeff` with an identity matrix
+        of appropriate size.
+
         Args:
-            num_electrons: the number of active electrons.
-            num_orbitals: the number of active orbitals.
-            num_alpha: the optional number of active alpha-spin electrons.
+            num_electrons: The number of active electrons. This may only be omitted if `freeze_core`
+                           is used.
+            num_molecular_orbitals: The number of active orbitals. This may only be omitted if
+                                    `freeze_core` is used.
+            num_alpha: The optional number of active alpha-spin electrons.
             active_orbitals: A list of indices specifying the molecular orbitals of the active
                              space. This argument must match with the remaining arguments and should
                              only be used to enforce an active space that is not chosen purely
                              around the Fermi level.
+            freeze_core: A convenience argument to quickly enable the inactivity of the
+                         `QMolecule.core_orbitals`. This keyword overwrites the use of all other
+                         keywords (except `remove_orbitals`) and, thus, cannot be used in
+                         combination with them.
+            remove_orbitals: A list of indices specifying molecular orbitals which are removed in
+                             combination with the `freeze_core` option. No checks are performed on
+                             the nature of these orbitals, so the user must make sure that these are
+                             _unoccupied_ orbitals, which can be removed without taking any energy
+                             shifts into account.
         """
         self.num_electrons = num_electrons
-        self.num_orbitals = num_orbitals
+        self.num_molecular_orbitals = num_molecular_orbitals
         self.num_alpha = num_alpha
         self.active_orbitals = active_orbitals
+        self.freeze_core = freeze_core
+        self.remove_orbitals = remove_orbitals
```

Comment:
Do we want these on public properties - private is fine. If they need to be made public then normally we have public getter/setter properties with docstrings.